mssqlcli/mssql_cli.py

```diff
@@ -390,15 +389,13 @@ def run(self):
 
         # raise error if interactive mode is set to false here
         if not self.interactive_mode:
-            raise ValueError("'run' must be used in interactive mode! Please set \
-                             interactive_mode to True.")
+            raise ValueError("Invalid arguments: 'run' must be used in interactive mode! Please set "
+                             "interactive_mode to True.")
 
         # exit and return error if user enters interactive mode with -o argument enabled
         if self.output_file:
-            click.secho("Invalid arguments: -o must be used with interactive mode set to false.",
-                        err=True, fg='red')
-            self.shutdown()
```

Comment:
Raising errors completes shutdown for us—I've validated that threads will not leak.